ps3.py

```diff
@@ -322,27 +322,41 @@ def play_game(word_list):
     word_list: list of lowercase strings
     """
     hands_number = int(input("Enter total number of hands: "))
-    hands_number_copy_for_cycle = hands_number
     score_for_the_game = 0
-    hand_match = deal_hand(HAND_SIZE)
-    replay_the_hand = 'yes'
+    replay_the_hand_switch = True
+    substitute_letter_switch = True
     while hands_number > 0:
+        hand_match = deal_hand(HAND_SIZE)
         # Main game
-        if hands_number != hands_number_copy_for_cycle:
-            # Needed for not asking user to replay the hand at first match
-            replay_the_hand = input("Would you like to replay the hand? ")
-        if replay_the_hand == 'no':
-            hand_match = deal_hand(HAND_SIZE)
-        if replay_the_hand == 'no' or hands_number == hands_number_copy_for_cycle:
+        if not substitute_letter_switch:
+            score_match = play_hand(hand_match, word_list)
+            score_for_the_game += score_match
+        else:
             display_hand(hand_match)
             substitute_answer = input("Would you like to substitute a letter? ")
             if substitute_answer == 'yes':
+                substitute_letter_switch = False
                 # Substituting letter in the hand, if user input is 'yes'
                 letter_substitute = input("Which letter would you like to replace: ")
-                hand_match = substitute_hand(hand_match, letter_substitute)
-        score_for_the_game += play_hand(hand_match, word_list)
-        hands_number -= 1
+                print()
+                hand_match_game_with_sub = substitute_hand(hand_match, letter_substitute)
+                score_match = play_hand(hand_match_game_with_sub, word_list)
+                score_for_the_game += score_match
+            else:
+                print()
+                score_match = play_hand(hand_match, word_list)
+                score_for_the_game += score_match
         print("--------")
+        if replay_the_hand_switch:
+            # Needed for not asking user to replay the hand at first match
+            replay_the_hand = input("Would you like to replay the hand? ")
+            if replay_the_hand == 'yes':
+                replay_the_hand_switch = False
+                score_for_the_game -= score_match
+                score_match_replay = play_hand(hand_match, word_list)
+                score_for_the_game += max(score_match, score_match_replay)
+                print("--------")
+        hands_number -= 1
```

Comment:
Please - implement it in easier way. From reading your code I can't understand in which cases you increase the score_for_the_game, and whether you are not doing it in case of replaying. Seems like you decided to call play_hand in many different scenarios separately - try to call it only once, that would be easier to read. Try make it more stupid.